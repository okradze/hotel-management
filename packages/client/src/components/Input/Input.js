import React from 'react'

const Input = ({ value, onChange, id, type, error, text }) => (
    <div className="LoginForm__input-group">
        <label htmlFor="name">{text}</label>
        <input
            id={id}
            value={value}
            onChange={onChange}
            className="input"
            type={type}
            autoComplete="off"
        />
        <p className="error SignupForm__error">{error && error}</p>
    </div>
)

export default Input
