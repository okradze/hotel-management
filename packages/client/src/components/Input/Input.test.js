import Input from './Input'

let value, onChange, id, type, error, text, wrapper

beforeEach(() => {
    value = ''
    onChange = jest.fn().mockImplementation(({ target }) => {
        value = target.value
    })
    text = 'Email'
    id = 'email'
    type = 'email'
    error = ''
    wrapper = shallow(
        <Input
            value={value}
            onChange={onChange}
            id={id}
            type={type}
            error={error}
            text={text}
        />,
    )
})

describe('Input', () => {
    it('renders', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('renders with error', () => {
        const errorText = 'Email is not valid'
        wrapper.setProps({ error: errorText })
        expect(wrapper.find('p').text()).toEqual(errorText)
        expect(wrapper).toMatchSnapshot()
    })

    it('calls function on onChange', () => {
        const inputValue = 'abc'
        const event = {
            target: {
                value: inputValue,
            },
        }
        wrapper.find('input').simulate('change', event)
        expect(onChange).toHaveBeenCalledWith(event)
        expect(value).toEqual(inputValue)
    })
})
