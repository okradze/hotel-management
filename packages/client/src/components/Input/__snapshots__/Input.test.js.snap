// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders 1`] = `
<div
  className="LoginForm__input-group"
>
  <label
    htmlFor="name"
  >
    Email
  </label>
  <input
    autoComplete="off"
    className="input"
    id="email"
    onChange={[MockFunction]}
    type="email"
    value=""
  />
  <p
    className="error SignupForm__error"
  />
</div>
`;

exports[`Input renders with error 1`] = `
<div
  className="LoginForm__input-group"
>
  <label
    htmlFor="name"
  >
    Email
  </label>
  <input
    autoComplete="off"
    className="input"
    id="email"
    onChange={[MockFunction]}
    type="email"
    value=""
  />
  <p
    className="error SignupForm__error"
  >
    Email is not valid
  </p>
</div>
`;
